### ISSUE TITLE
Where is the code source?

### ISSUE THREAD

**Issue by kimond** [extesy/DeckTracker#22]

**Comment by extesy** [comment]
I'm not ready yet to publish the entire source code.

**Comment by kimond** [comment]
I would like to work on a linux version if possible. To do this, I would like to know how do you proceed to communicate with the Legends application?

**Comment by extesy** [comment]
It injects a dll into a game process and then opens a TCP socket to communicate with it. I don't know how easy it would be to do on Linux. Then the main app is written in WPF which doesn't exist on Linux so it needs to be rewritten in some cross-platform UI framework.

**Comment by kimond** [comment]
Interesting. The Linux version might use a GTK interface or whatever Linux compatible libs through Wine rather than WPF. About the dll injected, Linux is likely to be able to do that using wine. I didn't validate my assumptions so I could investigate a little bit further.  Is it WPF that provides the overlays in the app?

**Comment by extesy** [comment]
No, the in-game overlay is using the standard Unity API so it would work in any OS that Unity supports. And non-Windows support requires two things: 1) dll injection, 2) off-game client UI.

**Comment by kimond** [comment]
Great, I think the workload would be not so big. Would I be able to look at the code to help me to figure out a solution or you're still unwilling?

**Comment by extesy** [comment]
I'll send you the source code later today to your email.

**Comment by kimond** [comment]
Thank you, I'll try to give you feedbacks as soon as possible!

**Comment by VitaliyKulikov** [comment]
@kimond are you running specific game under wine on linux OR **eternal** have native linux port? 
**elder scroll** don't have linux version ((

**Comment by kimond** [comment]
@VitaliyKulikov 
I'm running game with Steam on Wine. The game itself runs very well.

**Comment by VitaliyKulikov** [comment]
@kimond hmm, good news. i have tried few weeks ago, but had black screen - but game was running good with mouse moving with custom cursor and sounds working. seems, i should make clean prefix.
could you provide your configuration (wine version, etc.). thanks.

**Comment by kimond** [comment]
@VitaliyKulikov  I forgot to say that I need to run without my GPU (I have a laptop with bumblebee). If I run the game with the `primusrun` command I have a black screen with the sound working.

My config is:
Archlinux
Wine-2.12
I installed everything to make Steam wine working then Legends worked out of the box.

**Comment by VitaliyKulikov** [comment]
@kimond thanks. now it's better .. i am using NVIDIA Prime (Ubuntu 17.04) so i can switch to Intel too. But. still no luck. On Intel i Can see Boot Image (up-side - it's funny) .. but Main menu become black again (( 
i will try later again with looking in logs. Have you see this https://github.com/extesy/DeckTracker/issues/85#issuecomment-316504682

**Comment by kimond** [comment]
@VitaliyKulikov yeah I saw the #85 and I think you made a great job with the list. I forgot to say that Eternal works very well too with Steam Wine.

**Comment by VitaliyKulikov** [comment]
finally can run `game` under `wine 2.13-staging` even with `nvidia`. also, `tracker` works well ))

**Comment by kimond** [comment]
You managed to tun the tracker on linux?

**Comment by VitaliyKulikov** [comment]
@kimond yep. just installed .NET 4.6.2

**Comment by kimond** [comment]
I think we should write a documentation about the setup on Linux

**Comment by extesy** [comment]
@kimond Well, since you are the only one who was able to successfully run deck tracker on Linux, it would be reasonable for you to write this guide as well. What do you think? Can you make a pull request for `README.md` with Linux instructions?

**Comment by deglavs** [comment]
MacOS version please! :)

**Comment by AquaticoBrasil** [comment]
Hi, there is a way to install at MacOS??

**Comment by extesy** [comment]
@AquaticoBrasil Only in a virtual machine. You can also try asking /u/FMBrazuca on reddit if he can share his MacOS port.
